diff --git a/src/kar.cpp b/src/kar.cpp
index 46ce7d4..4bf60e9 100644
--- a/src/kar.cpp
+++ b/src/kar.cpp
@@ -87,7 +87,7 @@ bool KAr::openArchive(QIODevice::OpenMode mode)
         return true;
     }
     if (mode != QIODevice::ReadOnly && mode != QIODevice::ReadWrite) {
-        setErrorString(tr("Unsupported mode %1").arg(mode));
+        setErrorString(tr("Unsupported mode %1").arg(mode.toInt()));
         return false;
     }
 
diff --git a/src/karchive.cpp b/src/karchive.cpp
index c61962e..c882d46 100644
--- a/src/karchive.cpp
+++ b/src/karchive.cpp
@@ -159,7 +159,7 @@ bool KArchive::open(QIODevice::OpenMode mode)
     }
 
     if (!d->dev->isOpen() && !d->dev->open(mode)) {
-        setErrorString(tr("Could not open device in mode %1").arg(mode));
+        setErrorString(tr("Could not open device in mode %1").arg(mode.toInt()));
         return false;
     }
 
@@ -205,7 +205,7 @@ bool KArchive::createDevice(QIODevice::OpenMode mode)
         }
         break; // continued below
     default:
-        setErrorString(tr("Unsupported mode %1").arg(d->mode));
+        setErrorString(tr("Unsupported mode %1").arg(d->mode.toInt()));
         return false;
     }
     return true;
diff --git a/src/krcc.cpp b/src/krcc.cpp
index 649ab2e..32db125 100644
--- a/src/krcc.cpp
+++ b/src/krcc.cpp
@@ -114,7 +114,7 @@ bool KRcc::openArchive(QIODevice::OpenMode mode)
         return true;
     }
     if (mode != QIODevice::ReadOnly && mode != QIODevice::ReadWrite) {
-        setErrorString(tr("Unsupported mode %1").arg(mode));
+        setErrorString(tr("Unsupported mode %1").arg(mode.toInt()));
         return false;
     }
 
